KerbalStuff/blueprints/profile.py

```diff
@@ -27,7 +27,8 @@ def view_profile(username: str) -> str:
             if not current_user.admin:
                 abort(403)
     match = FORUM_PROFILE_URL_PATTERN.match(user.forumUsername)
-    forum_url_username = match.groups()[1] if match else None
+    forum_url_prefix = 'https://' if match and not match.group('prefix') else None
```

Comment:
```suggestion
    if not match.group('prefix'):
        # Not saved
        user.forumUsername = 'https://' + user.forumUsername
    elif match.group('prefix') == 'http://':
        user.forumUsername = 'https://' + user.forumUsername[7:]
```
What do you think of this logic instead? Replaces both no-scheme and the `http://`-scheme case with `https://`.

We could also save the change to the database by calling `db.commit`, but in this case we should also do that in `/profile/.../edit`, it'd be odd to only do it in the "read-only" view, code layout-wise.

Reply:
I'd rather stay away from updating the db in this part. Ideally we would preserve whatever the user types in the field and then make it work nicely, whatever it is...

Reply:
> I'd rather stay away from updating the db in this part.

Sure we can just use temporary variables to make sure that's not the case

> Ideally we would preserve whatever the user types in the field and then make it work nicely, whatever it is...

I think replacing HTTP with HTTPS links is okay in terms of changing what users entered, it gives them a little bit more security when browsing away from SpaceDock. We already know that the forum works with HTTPS, and it would redirect HTTP to HTTPS anyway, so we might as well save them that round trip (and prevent their first request from being sent in the clear, with e.g. the referer header, or even some cookies).

Reply:
OK, I had to move the concatenation logic out of the template to be able to track this in my head, but I think it replaces `http://` now and is somewhat cleaner.

Reply:
Thanks! Yeah I think having the concatenation outside of the template is better for readability, it doesn't have anything to do with the templating itself.